src/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -269,11 +280,51 @@ public string Host
         {
             get
             {
-                throw new PlatformNotSupportedException();
+                Uri hostUri = _hostUri ?? Address;
+                return (_hostUri == null || !_hostHasPort) && Address.IsDefaultPort ?
+                    hostUri.Host :
+                    hostUri.Host + ":" + hostUri.Port;
             }
             set
             {
-                throw new PlatformNotSupportedException();
+                if (RequestSubmitted)
+                {
+                    throw new InvalidOperationException(SR.net_writestarted);
+                }
+                if (value == null)
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+
+                Uri hostUri;
+                if ((value.IndexOf('/') != -1) || (!TryGetHostUri(value, out hostUri)))
+                {
+                    throw new ArgumentException(SR.net_invalid_host, nameof(value));
+                }
+
+                _hostUri = hostUri;
+
+                // Determine if the user provided string contains a port
+                if (!_hostUri.IsDefaultPort)
+                {
+                    _hostHasPort = true;
+                }
+                else if (value.IndexOf(':') == -1)
+                {
+                    _hostHasPort = false;
+                }
+                else
+                {
+                    int endOfIPv6Address = value.IndexOf(']');
+                    if (endOfIPv6Address == -1)
+                    {
+                        _hostHasPort = true;
+                    }
+                    else
+                    {
+                        _hostHasPort = value.LastIndexOf(':') > endOfIPv6Address;
+                    }
```

Comment:
Nit: this could be simplified to:
```C#
int endOfIPv6Address = value.IndexOf(']');
_hostHasPort = endOfIPv6Address == -1 || value.LastIndexOf(':') > endOfIPv6Address;
```